lib/chef/provider/package/yum.rb

```diff
@@ -51,17 +51,6 @@ def yum_binary
         # Extra attributes
         #
 
-        def arch_for_name(n)
-          if @new_resource.respond_to?("arch")
-            @new_resource.arch
-          elsif @arch
```

Comment:
I don't think this block gets entered since @new_resource should always respond to arch.